stringsubstitutor/src/main/java/org/jdbi/v3/stringsubstitutor/StringSubstitutorTemplateEngine.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.stringsubstitutor;
+
+import org.apache.commons.text.StringSubstitutor;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.TemplateEngine;
+
+@FunctionalInterface
+public interface StringSubstitutorTemplateEngine extends TemplateEngine {
+    StringSubstitutorTemplateEngine DEFAULTS = substitutor -> {};
+
+    void customize(StringSubstitutor substitutor);
```

Comment:
What do you anticipate is the use case for this method?

I feel like this would be clearer if this interface were a class instead, and the customize function was an optional `Consumer<StringSubstitutor>` constructor parameter.